quantus/metrics/axiomatic_metrics.py

```diff
@@ -277,7 +281,9 @@ def __init__(self, *args, **kwargs):
         self.perturb_func = self.kwargs.get(
             "perturb_func", baseline_replacement_by_indices
         )
-        self.perturb_baseline = self.kwargs.get("perturb_baseline", "black")
+        self.perturb_baseline = self.kwargs.get(
+            "perturb_baseline", "black"
+        )  # TODO: this attribute is unused right? also in other metrics
```

Comment:
True, please remove

Reply:
Upon investigation, this seems to be required due to the 

```
# Update kwargs.
self.kwargs = {
    **kwargs,
    **{k: v for k, v in self.__dict__.items() if k not in ["args", "kwargs"]},
}
```

in the `__call__` method

Reply:
Hmm but we don't use the attribute

Reply:
well we do through the self.__dict__.items()... although I agree that it is a bit weird

Reply:
hmm but not sure I get this one, `perturb_baseline` is never used in this metric am I right? then we remove it from being a `kwargs` key I would say

Reply:
`perturb_baseline` is used in the metric as a default for the perturbation. It is kind of invisible since it is passed through the self.kwargs attribute.
The issue is as follows (and how this is implemented is quite intransparent, we may want to think of a good way to change it; although I think setting different kwargs for perturb_func etc. as we planned should solve it too):

The line in the `__init__`

```
self.perturb_baseline = self.kwargs.get("perturb_baseline", "black")
```

sets `"perturb_baseline"` as a class attribute. Then, in the `__call__` method,  `self.kwargs` is overwritten by 

```
self.kwargs = {
            **kwargs,
            **{k: v for k, v in self.__dict__.items() if k not in ["args", "kwargs"]},
        }
```
where `self.kwargs` is set to the `kwargs` passed to `__call__`, with the following two caveats:
1.  All class attributes are turned into a dict and put into `kwargs`. This means that e.g. `perturb_baseline` will only exist in `self.kwargs` if it was a class attribute at that point or passed to `__call__` explicitly. I.e., if we do not set the class attribute in `__init__`, there won't be a default perturbation, instead errors will be thrown.
2. Due to the ordering of the two lines, parameters in the `kwargs` passed to `__call__` will be OVERWRITTEN by any class attributes with the same name, i.e., the default parameterization. This point is prone to cause very unexpected behavior, and we should probably change that^^